### ISSUE TITLE
ZhangSuenSkeletonization throws a StackOverflowException on large images

### ISSUE THREAD

**Issue by IsBrad** [accord-net/framework#1267]
**What would you like to submit?** (put an 'x' inside the bracket that applies)

  - [ ] question
  - [x] bug report
  - [ ] feature request

**Issue description**

The **ZhangSuenSkeletonization** class will cause a **StackOverflowException** when running on large(ish) images
This is due to a byte array almost the size of the input image being allocated on the stack.
.NET applications have a default stack size limit of 1mb for 32bit applications and 4mb for 64 bit.
This results in 1026x1026 & 2050x2050 almost exceeding the maximum stack size

The fix is to change this: (line 153)
```C#
var del0 = stackalloc byte[delSize]; for (var i = 0; i < delSize; i++) del0[i] = 0xFF;
```
To this:
```C#
var del0 = (byte*)Marshal.AllocHGlobal(delSize).ToPointer();
```

From testing this does not incur any measurable time penalty

**Comment by IsBrad** [comment]
I will submit pull request up with these fixes.

**Comment by justingruenberg** [comment]
this appears to have been merged in #1796 and can be closed.

**Comment by cesarsouza** [comment]
Thanks!
